pkg/snet/stun_handler.go

```diff
@@ -0,0 +1,386 @@
+package snet
+
+import (
+	"context"
+	"fmt"
+	"github.com/scionproto/scion/pkg/stun"
+	"golang.org/x/sync/errgroup"
+	"net"
+	"net/netip"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+)
+
+const timeoutDuration = 5 * time.Minute
+
+// stunHandler is a wrapper around net.UDPConn that handles STUN requests.
+type stunHandler struct {
```

Comment:
This is a connection, we should name it accordingly: `stunHandlerConn` or just `stunConn`?

Reply:
IMO `stunConn` is good